DRAFT_RELEASE.md

```diff
@@ -0,0 +1,45 @@
+Draft Release Notes: v1.14.0
```

Comment:
How much benefit do we get from this being a committed file that we edit during the release cycle (with PRs etc). I'm wondering if something else GitHub-ey would be sufficient (e.g. a tracking issue, wiki page, discussion, etc).

Reply:
The issue with a non versioned approach is that people can not contribute to it easily, external contributors would have to go through us to update the draft release notes or be granted the required permissions to edit them. 

Another option that I see would be to still commit it to the repo for visibility and to leverage the PR process we are used to, but instead of adding it to the PR checklist and expecting authors do that, we could take it as a recurring task for us to review merged PRs and update the draft release notes, so that we can avoid having the single person cutting the release also having to go through the 100s PRs and trying to spot relevant changes on the day of the release. WDYT?

Reply:
How about that we use some labeling conventions on PR, signaling if the merge is worth of showing up in the release notes? There are github actions like https://github.com/marketplace/actions/release-changelog-builder that look very powerful. In that way, we do not need to keep any release file around in the repo.